### PR TITLE
CD 804xx relnotes CDS-7758

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request

- Preview draft: [Latest: Version 804xx](https://64ec9dec70ea1b0ffb4260c3--harness-developer.netlify.app/release-notes/continuous-delivery)
-  [NG Release SAAS August 23rd 2023 for 804xx- Build](https://harness.atlassian.net/wiki/spaces/PD/pages/21476737025/NG+Release+SAAS+August+23rd+2023+for+804xx-+Build)

If tied to an Issue, list the Issue(s) here:

* [CDS-77758 CD release notes :: 804xx](https://harness.atlassian.net/browse/CDS-77758)

### CODE REVIEW

File: release-notes/continuous-delivery.md [hunk 1]
```diff
@@ -35,38 +35,55 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+- Fixed an issue where the K8s Apply step did not correctly evaluate expressions for the **Command Type** field. (CDS-76632)
 
-- Fixed an issue that caused the UI to crash when the input value of a component was changed from runtime to expression. (CDS-76216)	
+- Fixed an issue where the reconciliation for input set templates did not check for differences in old vs. new YAML for a given input set. This was happening because there was no flag to monitor for changes in YAML strings. (CDS-76533).
 
-- Fixed an issue with the Terraform Rollback step. The step did not inherit the backend configuration from the Terraform Apply step with the same provisioner Id, and you couldn't configure the backend configuration in the UI. This behavior was related to an issue where the rollback could fail when back-end configurations were not saved for the next run. (CDS-76071, ZD-48374)	
+  This fix introduces a new `yamlDiffPresent` boolean in `InputSetYamlDiffDTO` and its relevant unit tests. The purpose of this flag is similar to validYAML, which determines whether the current YAML is valid or not.
```
Comment (deepakbolangady) [inline release-notes/continuous-delivery.md:60]: This description does not have a Jira ID. Also, it mentions unit tests. I suspect it is exposed to users, but the mention of unit tests and the absence of a Jira ticket aroused my curiosity. Should we check? If it impacts users, like I think it does, we might have to rephrase it a bit.

File: release-notes/continuous-delivery.md [hunk 2]
```diff
@@ -35,38 +35,55 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+- Fixed an issue where users could not save a Manual Approval step as a template when `autoApprove : action` is set to `APPROVE` or `REJECT`.  (CDS-76350)
 
-- Fixed an issue with the **Repository Name** filter in the **Builds** page. Some users could not filter on builds that pulled their source code from a specific repository. If you experienced this issue, you will need to delete any saved filters that use the **Repository Name** filter and create them again. (CDS-75281, ZD-47876, ZD-48201)	
+- Added an information banner and doc link to the **Synchronization Barrier** step that highlights best practices when synchronizing deployments using barriers. (CDS-76291, ZD-48636)
+
+- Fixed a UI issue where the **Environment** tab in a Deploy stage did not display new custom variables after they were created.  (CDS-76068)
+
+- Removed the **Reconcile** option for individual input sets in the **Input Sets** page. (CDS-75845)
+
+- Implemented a fix to ensure that user-entered values are always populated in the **Inputs** tab on running / re-running the pipeline. (CDS-75593, ZD-48181)
```
Comment (deepakbolangady) [inline release-notes/continuous-delivery.md:74]:
```suggestion
- Implemented a fix to ensure that user-entered values are always populated on the **Inputs** tab when running or rerunning a pipeline. (CDS-75593, ZD-48181)
```

File: release-notes/continuous-delivery.md [hunk 3]
```diff
@@ -35,38 +35,55 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+
+- Fixed a filtering issue in the **Pipeline Executions** page. (CDS-75224)
```
Comment (deepakbolangady) [inline release-notes/continuous-delivery.md:76]:
```suggestion
- Fixed a filtering issue on the **Pipeline Executions** page. (CDS-75224)
```

File: release-notes/continuous-delivery.md [hunk 4]
```diff
@@ -35,38 +35,55 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+  
+  When the service received empty arrays for fields in the filter, the service would add this fields in the filtering criteria with empty values thereby not returning accurate results.
```
Comment (deepakbolangady) [inline release-notes/continuous-delivery.md:78]:
```suggestion
  If the service received empty arrays for fields in a filter, it added the fields to the filtering criteria with empty values, thereby not returning accurate results.
```
Better still (I'll begin my desc with "Services". Rewrite to "The service: if necessary, but we might have to clarify what service we are referring to):
> Services added fields to the filtering criteria even if they received those fields with empty arrays. Consequently, accurate results were not returned.

File: release-notes/continuous-delivery.md [hunk 5]
```diff
@@ -35,38 +35,55 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+  
+  The fix will handle the empty arrays being sent and saved filters will return only fields that have values in them.
```
Comment (deepakbolangady) [inline release-notes/continuous-delivery.md:80]:
```suggestion
  The fix handles empty arrays, and saved filters return only fields that have values.
```

File: release-notes/continuous-delivery.md [hunk 6]
```diff
@@ -35,38 +35,55 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+
+- Fixed an issue where ECR image based triggers were firing off with null values for artifact image tag and repo name. (CDS-75173)
```
Comment (deepakbolangady) [inline release-notes/continuous-delivery.md:82]:
```suggestion
- Fixed an issue where triggers based on ECR images were fired with null values for artifact image tag and repo name. (CDS-75173)
```
Should it be "even when the artifact image tag and repository name had null values"?

File: release-notes/continuous-delivery.md [hunk 7]
```diff
@@ -35,38 +35,55 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+
+- Added a tooltip for Build stages to let users know `CI Stage can be skipped with New Artifact/Manifest Trigger using selective stage configuration`. (CDS-75080, ZD-47902)
```
Comment (deepakbolangady) [inline release-notes/continuous-delivery.md:84]:
```suggestion
- Added the following tooltip for the Build stage: `CI Stage can be skipped with New Artifact/Manifest Trigger using selective stage configuration`. (CDS-75080, ZD-47902) 
```

**Review by deepakbolangady — APPROVED** [review]
Approving with some comments/questions.

File: release-notes/continuous-delivery.md (hunk 1 above)
Comment (douglas-j-bothwell) [reply release-notes/continuous-delivery.md:60]: This is one paragraph in a two-paragraph summary. Sandesh has approved my [summary draft](https://harness.atlassian.net/browse/CDS-76533?focusedCommentId=560120), I'm going to leave this as-is.

**Review by prasadsatamharness — DISMISSED** [review]

**Review by deepakbolangady — APPROVED** [review]

**Pull request merged**